plugins/inputs/opcua/read_client.go

```diff
@@ -55,18 +60,26 @@ func (rc *readClientConfig) createReadClient(log telegraf.Logger) (*readClient,
 		rc.ReadRetryTimeout = config.Duration(100 * time.Millisecond)
 	}
 
+	// Set default for ReconnectErrorThreshold if not configured
+	reconnectThreshold := rc.ReconnectErrorThreshold
+	if reconnectThreshold == 0 {
+		reconnectThreshold = 1 // Default value
+	}
+
```

Comment:
I don't think this is necessary as a threshold of zero should work the same way as one...